node/tests/st/bgp/test_switch_route_reflector.py

```diff
@@ -0,0 +1,151 @@
+            node3cfg['metadata']['labels'] = {
+                'routeReflectorClusterID': node3cfg['spec']['bgp']['routeReflectorClusterID'],
+            }
+            host3.add_resource(node3cfg)
+
+            # Configure peerings - note, NOT a full mesh - from the
+            # other nodes to the route reflector.
+            host3.add_resource({
+                'apiVersion': 'projectcalico.org/v3',
+                'kind': 'BGPPeer',
+                'metadata': {
+                    'name': 'rr-peerings',
+                },
+                'spec': {
+                    'nodeSelector': '!has(routeReflectorClusterID)',
+                    'peerSelector': 'has(routeReflectorClusterID)',
+                },
+            })
+
+            # Wait until the peers' BGP session to the route reflector are
+            # established
+            retry_until_success(check_bird_status, 30, AssertionError,
+                                host1, [("node specific", host3.ip, "Established")])
+            retry_until_success(check_bird_status, 30, AssertionError,
+                                host2, [("node specific", host3.ip, "Established")])
+
+            # Turn off the node-to-node mesh (do this from any host).
+            update_bgp_config(host3, nodeMesh=False)
+
+            # Allow network to converge
+            self.assert_true(workload_host1.check_can_ping(workload_host2.ip, retries=20))
+
+            # Verify results of continuous ping from host1's to host2's workloads
+            # and vice-versa
+            t1.close()
+            t1.join()
+            logger.info("ping1_result: %s", str(ping1_result.get()))
+            self.assert_true(ping1_result.get())
+            t2.close()
+            t2.join()
+            logger.info("ping2_result: %s", str(ping2_result.get()))
+            self.assert_true(ping2_result.get())
+
+            # Check connectivity in both directions.
+            self.assert_ip_connectivity(workload_list=[workload_host1,
+                                                       workload_host2],
+                                        ip_pass_list=[workload_host1.ip,
+                                                      workload_host2.ip],
+                                        retries=5)
+
+    @attr('slow')
+    def test_bird_switch_route_reflector(self):
+        self._test_switch_route_reflector(backend='bird')
+
+    @attr('slow')
+    def test_bird_switch_route_reflector_default_as(self):
+        self._test_switch_route_reflector(backend='bird', bgpconfig_as_num=None, peer_as_num=64512)
+
+    # TODO: Add back when gobgp is updated to work with libcalico-go v3 api
+    @attr('slow')
+    @skip("Disabled until gobgp is updated with libcalico-go v3")
+    def test_gobgp_switch_route_reflector(self):
+        self._test_switch_route_reflector(backend='gobgp')
+
+TestSwitchRouteReflector.batchnumber = 1  # Adds a batch number for parallel testing
```

Comment:
I'm not quite sure if I missed where this gets used. At first I thought it was a requirement for running `apply_async` but after I took a look back, I realized I didn't quite understand how this gets used.

Reply:
TBH I think this was in the file I based myself on for this test (node/tests/st/bgp/test_internal_route_reflector.py), I looked around and it's used by https://github.com/projectcalico/calico/blob/9e10c8b0aef36e7727c69b58a371360a77b4c820/node/tests/st/test_base.py#L56 to but I couldn't understand 100% how it works...

Reply:
I see. It looks like we're just assigning a batchnumber for the test to run in. I guess it should be fine.